### ISSUE TITLE
Configure Catalyst to use catalyst-uploader by default

### ISSUE THREAD

**Issue by thomshutt** [livepeer/go-livepeer#2506]
Design: https://www.notion.so/livepeer/Catalyst-Q3-2022-4dc6314de43f4a3ab916b5f0923b7f45

This task encapsulates including build / bundling of the dms-uploader project as part of the Catalyst build and then whatever config is required to make sure Mist uses it to upload all incoming segments

**Comment by cyberj0g** [comment]
#### Current progress
- [catalyst-uploader](https://github.com/livepeer/catalyst-uploader/) repository is created, CI/CD is configured
- AWS S3 support is implemented in a separate [go-tools](https://github.com/livepeer/go-tools) package, which is shared with [go-livepeer](https://github.com/livepeer/go-livepeer)
- catalyst-uploader binary is [added](https://github.com/livepeer/catalyst/pull/115) to Catalyst

#### Next steps
- Spec out following unclear parts:
    > by default

    > whatever config is required to make sure Mist uses it to upload all incoming segments
- Implementation of above

**Comment by cyberj0g** [comment]
I have tested two approaches to implementing automatic live stream upload to S3 with Mist + Catalyst:
##### 1. Mist `PUSH_START` API call after stream creation
The function is described in section 7.1.33 of the [manual](https://mistserver.org/guides/MistServer_Manual_3.0.pdf). It takes stream name and destination URI parameter, and then invokes `catalyst-uploader` for `s3://` scheme. Naturally, something should make this API call right after creating a new stream. I think, it can be added to `catalyst-api`.
##### 2. Wildcard automatic push in `mistserver.conf`
This one is closer to literal 'upload all incoming segments of live streams to S3 by default' requirement. It requires a change [here](https://github.com/DDVTECH/mistserver/blob/0af992d4057cef8ebef53eb67be809c713262af2/src/controller/controller_push.cpp#L371) to match any stream without specific prefix. And update to [values](https://github.com/livepeer/livepeer-infra/blob/master/charts/catalyst/values.yaml#L60) of Helm chart, which generates Mist config, to include following section:
```
{
...
  "autopushes": [
    [
      "",
      "s3+http://REDACTED:REDACTED@localhost:9000/bucket/$stream/playlist.m3u8"
    ]
  ],
}
```

Both approaches are viable and fairly straightforward to implement. The main question is how to properly parametrize and manage configuration of this S3 upload feature. I see following parameters:
- **On/Off switch** - that's the main reason to prefer option 1., in my opinion. Even if we won't implement this switch in the first version, if it's required later, it can be easily added to the code making explicit `PUSH_START` call. On the contrary, if we go with option 2. now, we'll have to replace the implementation with something resembling 1. to have this switch.
- **Target upload URI** consisting of: key, secret, host, port, bucket, stream_id, object_key
Stream_id is dynamic and unique (Ethereum address). It's a `$stream` macro in option 2. The rest of parameters can be put to `livepeer_infra`, if they not supposed to be user-configurable.

To summarize, I would go with option 1., because 2. feels like adding to technical debt. Would appreciate an input on where in `catalyst-api` to add that API call.

WDYT @thomshutt 

##### Side note
To stream directly into Mist bundled with Catalyst, one need to first remove `PUSH_REWRITE` trigger from Mist config. If using `make dev`, it is located at `~/.config/livepeer/mistserver.dev.conf`. After that, streaming with Ffmpeg works as expected:
```
ffmpeg -re -i bbb.mp4 -c:a copy -c:v copy -f flv rtmp://localhost/live/test_stream
```

**Comment by cyberj0g** [comment]
Decided: go with option 2, because it's most straightforward. No need to modify Mist autopush matching, because livepeer PUSH_REWRITE trigger generates stream name starting with `video+`, and it can already be matched by `video+` wildcard.
